domain-models-runtime/src/main/java/org/folio/rest/persist/PostgresClient.java

```diff
@@ -2537,6 +2538,41 @@ public void select(String sql, Handler<AsyncResult<ResultSet>> replyHandler) {
     client.getConnection(conn -> select(conn, sql, closeAndHandleResult(conn, replyHandler)));
   }
 
+  static void queryAndAnalyze(SQLConnection conn, String sql, String statMethod,
+    Handler<AsyncResult<ResultSet>> replyHandler) {
+
+    long start = System.nanoTime();
+    conn.query(sql, res -> {
+      long queryTime = (System.nanoTime() - start);
+      StatsTracker.addStatElement(STATS_KEY + statMethod, queryTime);
+      if (res.failed()) {
+        log.error("queryAndAnalyze: " + res.cause().getMessage() + " - "
+          + sql, res.cause());
+        replyHandler.handle(Future.failedFuture(res.cause()));
+        return;
+      }
+      if (queryTime >= explainQueryThreshold * 1000000) {
```

Comment:
Can you reduce indentation and complexity and improve readability by using a guard clause?

    if (queryTime < explainQueryThreshold * 1000000) {
      replyHandler.handle(res);
      return;
    }
    final String explainQuery = "EXPLAIN ANALYZE " + sql;
    …

Reply:
I'm going to say no to that one.. Why? SQ did not complain and guard clauses IMHO are mostly for exceptional cases/errors.. which this is NOT.

Reply:
SQ doesn't have any guard clause rules, this is checked by human code review only.
Guard clauses are *not* mostly for exceptional cases/errors:
* https://refactoring.guru/replace-nested-conditional-with-guard-clauses
* http://wiki.c2.com/?GuardClause
They are also for checking the state of the object or checking for trivial cases.
Can you elaborate whether that guard clause reduces complexity and improves readability?

Reply:
Using the proposed guard clause reduces the Cognitive Complexity (as calculated by SQ) by 4.